de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/ad/session/MFSession.java

```diff
@@ -226,10 +228,31 @@ public String getHostKey(final Properties ctxToUpdate)
 		{
 			return hostKey; // the host key was already determined. Nothing more to do.
 		}
+		
+		final String newHostKey;
+		final IHostKeyBL hostKeyBL = Services.get(IHostKeyBL.class);
+		// check user for hostkey
+		final int userId = Env.getAD_User_ID(ctxToUpdate);
+		if (userId > 0)
```

Comment:
instead of having these ifs and elses here, they should rather be somehwere behind hostKeyBL.getCreateHostKey()